File: easy/reverse_vowels.py
def is_vowel(s: str) -> bool:
    return s.lower() in ("a", "e", "i", "o", "u")


def reverse_vowels(s: str) -> str:
    left, right, result = 0, len(s) - 1, [""] * len(s)
    while left <= right:
        if not is_vowel(s[left]):
            result[left] = s[left]
            left += 1
        elif not is_vowel(s[right]):
            result[right] = s[right]
            right -= 1
        else:
            result[right] = s[left]
            result[left] = s[right]
            right -= 1
            left += 1

    return "".join(result)

File: easy/reverse_vowels.go
package easy

func IsVowel(r rune) bool {
	switch r {
	case 'a', 'A', 'e', 'E', 'i', 'I', 'o', 'O', 'u', 'U':
		return true
	default:
		return false
	}
}
func ReverseVowels(s string) string {
	runes := []rune(s)
	left, right := 0, len(runes)-1
	for left < right {
		for left < right && !IsVowel(runes[left]) {
			left++
		}
		for left < right && !IsVowel(runes[right]) {
			right--
		}
		runes[left], runes[right] = runes[right], runes[left]
		left++
		right--
	}
	return string(runes)
}
